docs/3.0.0-beta.x/guides/unit-testing.md

```diff
@@ -0,0 +1,312 @@
+
+`Supertest` allows you to test all the `api` routes as they were instances of [http.Server](https://nodejs.org/api/http.html#http_class_http_server)
+
+:::: tabs
+
+::: tab yarn
+`yarn add jest supertest`
+:::
+
+::: tab npm
+`npm install jest supertest`
+:::
+
+Once this is done add this to `package.json` file
+
+add `test` command to `scripts` section
+
+```
+  "scripts": {
+    "develop": "strapi develop",
+    "start": "strapi start",
+    "build": "strapi build",
+    "strapi": "strapi",
+    "test": "jest --detectOpenHandles"
+  },
+```
+
+and add those line at the bottom of file
+
+```
+  "jest": {
+    "testPathIgnorePatterns": [
+      "/node_modules/",
+      ".tmp",
+      ".cache"
+    ]
+  }
+```
+
+Those will inform `Jest` not to look for test inside the folder where it shouldn't.
+::::
+
+## Introduction
+
+### Testing environment
+
+Test framework must have a clean empty environment to perform valid test and also not to interfere with current database.
+
+Once `jest` is running it uses the `test` [enviroment](../concepts/configurations.md#environments) (switching `NODE_ENV` to `test`)
+so we need to create a special environment setting for this purpose.
+Copy contents of `./config/environments/development` into `./config/environments/test` and add some changes.
+Edit `./config/environments/test/database.json` and add value `"filename": ".tmp/test.db"` - the reason of that is that we want to have a separate sqlite database for tests, so our test will not touch real data.
+This file will be temporary, each time test is finished, we will remove that file that every time tests are run on the clean database.
+The whole file will look like this:
+
+**Path —** `./config/environments/test/database.json`
+
+```json
+{
+  "defaultConnection": "default",
+  "connections": {
+    "default": {
+      "connector": "bookshelf",
+      "settings": {
+        "client": "sqlite",
+        "filename": ".tmp/test.db"
+      },
+      "options": {
+        "useNullAsDefault": true,
+        "pool": {
+          "min": 0,
+          "max": 15
+        }
+      }
+    }
+  }
+}
+```
+
+### Strapi instance
+
+In order to test anything we need to have a strapi instance that runs in the testing eviroment,
+bascially we want to get instance of strapi app as object, similar like creating an instance for [process manager](process-manager).
+
+These tasks require adding some files - let's create a folder `tests` where all the tests will be put and inside it, next to folder `helpers` where main Strapi helper will be in file strapi.js.
+
+**Path —** `./tests/helpers/strapi.js`
+
+```js
+const Strapi = require('strapi');
+const http = require('http');
+
+let instance;
+
+async function setupStrapi() {
+  if (!instance) {
+    /** the follwing code in copied from `./node_modules/strapi/lib/Strapi.js` */
+    await Strapi().load();
```

Comment:
we made some changes to the load function, it now calls the bootstrap directly

Reply:
@alexandrebodin - Which of https://www.npmjs.com/package/strapi versions I should use in my example (https://github.com/qunabu/strapi-unit-test-example/blob/master/package.json) in order to use up to date version? 

I'm using latest `3.0.0-beta.20.3`

Reply:
You can use 3.0.0 :D

Reply:
@alexandrebodin I've updated my example code https://github.com/qunabu/strapi-unit-test-example 

I'm going to update this as well, yet should this PR be into `master` or `documentation` branch?